### PR TITLE
3fv3UwKG: Support uploading a certificate

### PR DESCRIPTION
**Pull request by AmritSidhu**
Created a model with a table that has two fields, certType & certificate
Created an upload view for the form and to print out database data
Implemented Gov styling for the form and outputted data
Created model test for some validation
Renamed HomeController to CertificatesController

### CODE REVIEW

File: app/views/certificates/upload.html.erb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    <h1 class="govuk-fieldset__heading">Upload your certificate</h1>
+  </legend>
+
+  <div class="govuk-form-group">
+    <%= form_for @upload, url: '/upload'  do |f| %>
+      <h3 class="govuk-heading-m">Certificate usage</h3>
+      <div class="govuk-radios govuk-radios--inline">
+        <div class="govuk-radios__item">
+          <%= f.radio_button :usage, 'Signing', class: 'govuk-radios__input'  %>
+          <%= f.label :usage, 'Signing', :value => 'Signing', class: 'govuk-label govuk-radios__label' %>
```
Comment (jakubmiarka) [inline app/views/certificates/upload.html.erb]: I would use lowercase value, i.e. `signing`

File: app/models/certificate.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class Certificate < ApplicationRecord
+  validates_presence_of :usage
```
Comment (jakubmiarka) [inline app/models/certificate.rb]: we could do `validates_inclusion_of :usage, in: ['signing', 'encryption']` just to start with at least... the actual cert validation is covered by another story

File: app/views/certificates/upload.html.erb (hunk 1 above)
Comment (AmritSidhu) [reply app/views/certificates/upload.html.erb]: Is there any need for the value under label?

File: app/models/certificate.rb (hunk 2 above)
Comment (parthagds) [reply app/models/certificate.rb]: validates_inclusion_of: Validates whether the value of the specified attribute is available in a particular enumerable object. 

However I think we want to check whether or not these 2 attributes are null (which is in the scope of this card?).

Comment (parthagds) [reply app/models/certificate.rb]: Would it be like this?
validates_inclusion_of :usage, in: ['signing', 'encryption']
validates_presence_of :value, message: 'You must enter certificate value.'

Comment (jakubmiarka) [reply app/models/certificate.rb]: the inclusion validates for null (i.e. if it's not one of the options it fails the validation). We should still keep the null validation on `value` on which we'll build with the next story

File: spec/models/certificate_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe Certificate, type: :model do
+  it "is valid with valid attributes" do
+    expect(Certificate.new(usage: 'signing', value: 'Test')).to be_valid
```
Comment (jakubmiarka) [inline spec/models/certificate_spec.rb]: would also be good to test when the usage is `blah` i.e. check we're not allowing non-allowed usage types

File: spec/models/certificate_spec.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    expect(Certificate.new(usage: 'encryption', value: 'Test')).to be_valid
+  end
+
+  it "is not valid with non-valid attributes" do
+    expect(Certificate.new(usage: 'blah', value: 'Test')).to_not be_valid
+  end
+
+  it "is not valid without a usage and/or value" do
+    expect(Certificate.new(usage: nil)).to_not be_valid
+    expect(Certificate.new(value: nil)).to_not be_valid
```
[the remark this answers is not in the document]
Comment (AmritSidhu) [reply spec/models/certificate_spec.rb]: Doesn't the first test `is valid with valid attributes` cover that

[the remark this answers is not in the document]
Comment (jakubmiarka) [reply spec/models/certificate_spec.rb]: from what I can see, it'll let you submit an empty certificate (as long as it has a correct type)?

**Pull request merged**